tools/flat-storage/src/commands.rs

```diff
@@ -258,10 +373,17 @@ impl FlatStorageCommand {
         opener: StoreOpener,
     ) -> anyhow::Result<()> {
         let (_, epoch_manager, rw_hot_runtime, rw_chain_store, rw_hot_store) =
-            Self::get_db(&opener, home_dir, &near_config, near_store::Mode::ReadWriteExisting);
+            Self::get_init_db(&opener, home_dir, &near_config, cmd.height, &cmd.target_db_path);
 
         let tip = rw_chain_store.final_head()?;
         let shard_uid = epoch_manager.shard_id_to_uid(cmd.shard_id, &tip.epoch_id)?;
+
+        let mut store_update = rw_hot_store.store_update();
+        store_update
+            .set_ser(DBCol::FlatStorageStatus, &shard_uid.to_bytes(), &FlatStorageStatus::Empty)
+            .expect("Unable to set FS status");
+        store_update.commit().expect("Unable to change FS status");
+
         let mut creator =
             FlatStorageShardCreator::new(shard_uid, tip.height - 1, epoch_manager, rw_hot_runtime);
```

Comment:
There's some weird logic checking if specific deltas exist for the start_height parameter. I would probably try to set this to `tip.height` instead of `tip.height - 1` to begin with, although @Longarithm or @pugachAG might have a better idea why this has a `-1`

Reply:
Maybe for more safety... It is easier to move head forward later than backwards (see `MoveFlatHeadMode::Back`). `tip.height` also works.